### PR TITLE
Implement Accelerate DMA for Texture Cache and correct some methods in DMA engine

### PR DESCRIPTION
**Pull request by FernandoS27**
This PR takes care of providing a safe conduct to do DMAs on the GPU if any of the textures is within the Texture Cache. It also corrects some missing things in Linear->Tile Copies and Tiled->Linear Copies as a result DKCTP now displays text and icons correctly.

### CODE REVIEW

File: src/video_core/texture_cache/texture_cache.h [hunk 1]
```diff
@@ -773,6 +843,67 @@ class TextureCache {
...
+
+    std::pair<TSurface, TView> GetDMASurface(
+        const Tegra::Engines::MaxwellDMA::SurfaceConfig& config, const DMAInfo& info) {
+        SurfaceParams params = SurfaceParams::CreateForDMASurface(config, info.component_type,
+                                                                  info.pixel_format, info.target);
+        const GPUVAddr gpu_addr = config.gpu_addr;
+        return GetSurface(gpu_addr, params, true, false);
+    }
+
+    SurfaceTarget FigureDMATarget(const Tegra::Engines::MaxwellDMA::SurfaceConfig& config) {
```
Comment (lioncash) [inline src/video_core/texture_cache/texture_cache.h]:
```suggestion
    SurfaceTarget DetermineDMATarget(const Tegra::Engines::MaxwellDMA::SurfaceConfig& config) {
```

File: src/video_core/rasterizer_interface.h [hunk 2]
```diff
@@ -22,6 +23,13 @@ enum class LoadCallbackStage {
     Build,
     Complete,
 };
+
+enum Caches : u32 {
```
Comment (lioncash) [inline src/video_core/rasterizer_interface.h]: `enum class` and provide the relevant operations. This should also be named something like `CacheFlags`

File: src/video_core/renderer_opengl/gl_rasterizer.cpp [hunk 3]
```diff
@@ -811,6 +811,14 @@ void RasterizerOpenGL::FlushAndInvalidateRegion(CacheAddr addr, u64 size) {
     InvalidateRegion(addr, size);
 }
 
+u32 RasterizerOpenGL::IsCacheHit(const GPUVAddr gpu_addr, const std::size_t size) {
```
Comment (lioncash) [inline src/video_core/renderer_opengl/gl_rasterizer.cpp]: This should return the actual type of the flags and not erase the type into a u32

File: src/video_core/rasterizer_interface.h (hunk 2 above)
Comment (FernandoS27) [reply src/video_core/rasterizer_interface.h]: I can't use enum class as a flag. It does not inherit the Bitwise Or operator.

**Comment by FernandoS27** [comment]
I've learned that accelerate DMA is mostly unfeasible due to how it works and that oit's used to prepare a texture we got no info on. I've learned a few things on how this DMAs work and I'll add the corrections and relaxations to another PR.

**Pull request closed without merging**